File: frontend/src/api/myChannel.ts
/**
 * MyChannel API（v7.1 · MC-2～MC-6）
 * 使用 fetchAPIEnvelope：後端 envelope 含 data + balance/empty，不可被 interceptor 拆掉。
 */
import { fetchAPIEnvelope } from './client';

export interface MyChannelFeedCard {
  id: string;
  heading: string;
  intro: string;
  category?: string;
  image_url?: string | null;
}

export interface MyChannelFeedResponse {
  data: MyChannelFeedCard[];
  balance: number;
  lang: string;
  cached: boolean;
  rate_limited: boolean;
  empty: boolean;
  has_channels: boolean;
}

export interface UnlockResponse {
  topic_id: string;
  source_url: string;
  digest_300: string;
  balance: number;
}

export interface ChannelTemplate {
  id: string;
  category: string;
  region: string;
  name_key: string;
  desc_key: string;
  suggested_name: string;
}

export const myChannelApi = {
  getFeed: (lang: string) =>
    fetchAPIEnvelope<MyChannelFeedResponse>(
      `/my-channel/feed?lang=${encodeURIComponent(lang)}`
    ),

  unlock: (topicId: string, idempotencyKey: string) =>
    fetchAPIEnvelope<UnlockResponse>(`/my-channel/topics/${topicId}/unlock`, {
      method: 'POST',
      body: JSON.stringify({ idempotency_key: idempotencyKey }),
    }),

  getChannelTemplates: () =>
    fetchAPIEnvelope<{ data: ChannelTemplate[] }>('/my-channel/channel-templates'),
};

File: backend/app/schemas/my_channel.py
"""
MyChannel API schemas（MC-2～MC-5）
"""
from typing import List, Optional

from pydantic import BaseModel, Field


class MyChannelFeedCard(BaseModel):
    id: str
    heading: str
    intro: str = Field(..., max_length=30)
    category: Optional[str] = None
    image_url: Optional[str] = None


class MyChannelFeedResponse(BaseModel):
    data: List[MyChannelFeedCard]
    balance: int
    lang: str
    cached: bool = False
    rate_limited: bool = False
    empty: bool = False
    has_channels: bool = False


class ChannelTemplateItem(BaseModel):
    id: str
    category: str
    region: str
    name_key: str
    desc_key: str
    suggested_name: str


class ChannelTemplatesResponse(BaseModel):
    data: List[ChannelTemplateItem]


class UnlockResponse(BaseModel):
    topic_id: str
    source_url: str
    digest_300: str = Field(..., max_length=300)
    balance: int


class UnlockRequest(BaseModel):
    idempotency_key: str = Field(..., min_length=8, max_length=64)


class AddCreditsRequest(BaseModel):
    amount: int = Field(..., gt=0, le=1000)


class AddCreditsResponse(BaseModel):
    user_id: str
    balance: int
    added: int
